ipn/prefs.go

```diff
@@ -105,6 +105,12 @@ type Prefs struct {
 	ExitNodeID tailcfg.StableNodeID
 	ExitNodeIP netip.Addr
 
+	// ExitNodePrior is the most recently used ExitNodeID in string form. It is set by
+	// the backend on transition from exit node on to off and used by the
+	// backend. It's not of type tailcfg.StableNodeID because in the future we plan
+	// to overload this field to mean things like "Anything in country $FOO" too.
+	ExitNodePrior string
```

Comment:
How about an internal substruct

Reply:
I don't follow?

Reply:
Oh, a struct type here? That gets really annoying in Prefs. Look at all the pain AutoUpdates needed to go through to make it work.

Reply:
I just mean that maybe we should have an internal storage that's different from prefs, or at least a different section within prefs to make it clearer.

Reply:
Oh, yeah, I'd love that. I don't want this to be writable by LocalAPI clients.

I tried to read the profileManager code to figure out what the types it's actually writing to StateStore and figured I was getting sidetracked.

Should we do that first? Add some sibling to Prefs that's like ProfilePrivateMeta?

Reply:
Doing that first does seem like the appropriate thing to do.

Reply:
Per chat, updated to /^Internal/ prefs field for now, with protections+tests in place to block EditPrefs of an internal field via LocalAPI.